pipeline/node.go

```diff
@@ -327,8 +327,8 @@ func (n *chainnode) HttpOut(endpoint string) *HTTPOutNode {
 }
 
 // Creates an HTTP Post node that POSTS received data to the provided HTTP endpoint.
-func (n *chainnode) HttpPost(url string) *HTTPPostNode {
-	h := newHTTPPostNode(n.provides, url)
+func (n *chainnode) HttpPost(url ...string) *HTTPPostNode {
```

Comment:
We should document that the URL  is optional if the endpoint is specified, hence the variadic signature.